src/aiida_quantumespresso/common/hubbard.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+"""Utility class and functions for HubbardStructureData."""
+# pylint: disable=no-name-in-module, invalid-name
+from typing import List, Literal, Union
+
+from pydantic import BaseModel, conint, conlist, constr, validator
+
+formulation_names = Literal['dudarev', 'liechtenstein']
+type_names = Literal['Ueff', 'U', 'V', 'J', 'B', 'E2', 'E3']
+projectors_name = Literal['atomic', 'ortho-atomic', 'norm-atomic', 'wannier-functions', 'pseudo-potentials',]
+
+
+class HubbardParameters(BaseModel):
+    """Class for describing onsite and intersite Hubbard interaction parameters.
+
+    .. note: allowed manifold formats are:
+            * {N}{L} (2 characters)
+            * {N1}{L1}-{N2}{L2} (5 characters)
+
+        N = quantum number (1,2,3,...); L = orbital letter (s,p,d,f,g,h)
+
+    :param atom_index: atom index in the abstract structure
+    :param atom_manifold: atom manifold (syntax is `3d`, `3d-2p`)
+    :param neighbour_index: neighbour index in the abstract structure
+    :param neighbour_manifold: atom manifold (syntax is `3d`, `3d-2p`)
+    :param translation: translation vector referring to the neighbour atom, (3,) shape list of ints
+    :param value: value of the Hubbard parameter, expessed in eV
+    :param hubbard_type: type of the Hubbard parameters used (`Ueff`, `U`, `V`, `J`, `B`, `E2`, `E3`)
+    """
+
+    atom_index: conint(strict=True, ge=0)
+    atom_manifold: constr(strip_whitespace=True, to_lower=True, min_length=2, max_length=5)
+    neighbour_index: conint(strict=True, ge=0)
+    neighbour_manifold: constr(strip_whitespace=True, to_lower=True, min_length=2, max_length=5)
+    translation: conlist(conint(strict=True), min_items=3, max_items=3)
+    value: float
+    hubbard_type: type_names  # default to 'U' ?
```

Comment:
Since these are not really method parameters, but class attributes, the standard way to document them is directly below the attribute.
```suggestion
    atom_index: conint(strict=True, ge=0)
    """Atom index in the abstract structure."""

    atom_manifold: constr(strip_whitespace=True, to_lower=True, min_length=2, max_length=5)
    """Atom manifold (syntax is `3d`, `3d-2p`)."""

    neighbour_index: conint(strict=True, ge=0)
    """Neighbour index in the abstract structure."""

    neighbour_manifold: constr(strip_whitespace=True, to_lower=True, min_length=2, max_length=5)
    """Atom manifold (syntax is `3d`, `3d-2p`)."""

    translation: conlist(conint(strict=True), min_items=3, max_items=3)
    """Translation vector referring to the neighbour atom, (3,) shape list of ints."""

    value: float
    """Value of the Hubbard parameter, expessed in eV."""

    hubbard_type: type_names  # default to 'U' ?
    """Type of the Hubbard parameters used (`Ueff`, `U`, `V`, `J`, `B`, `E2`, `E3`)."""
```